packages/react-server/core/__tests__/context/navigator/NavigatorSpec.js

```diff
@@ -0,0 +1,140 @@
+		requestContext = new RequestContextStub(options);
+	});
+	afterEach(() => {
+		requestContext = null;
+	});
+
+	it("routes to a basic page using an unspecified method get", (done) => {
+		const req = {
+			method: "get",
+			protocol: "http",
+			secure: false,
+			hostname: "localhost",
+			url: "/basicPage",
+		};
+		const expressRequest = new ExpressServerRequest(req);
+
+		requestContext.navigator.on('page', () => {
+			expect(expressRequest.getRouteName() === 'BasicPage');
+			done();
+		});
+
+		requestContext.navigate(expressRequest, History.events.PAGELOAD);
+	});
+
+	it("routes to a basic page using an unspecified method HEAD", (done) => {
+		const req = {
+			method: "head",
+			protocol: "http",
+			secure: false,
+			hostname: "localhost",
+			url: "/basicPage",
+		};
+		const expressRequest = new ExpressServerRequest(req);
+
+		requestContext.navigator.on('page', () => {
+			expect(expressRequest.getRouteName() === 'BasicPage');
+			done();
+		});
+
+		requestContext.navigate(expressRequest, History.events.PAGELOAD);
+	});
+
+	it("routes to a page using a method GET (caps)", (done) => {
+		const req = {
+			method: "get",
+			protocol: "http",
+			secure: false,
+			hostname: "localhost",
+			url: "/basicPageCaps",
+		};
+		const expressRequest = new ExpressServerRequest(req);
+
+		requestContext.navigator.on('page', () => {
+			expect(expressRequest.getRouteName() === 'BasicPageCaps');
+			done();
+		});
+
+		requestContext.navigate(expressRequest, History.events.PAGELOAD);
+	});
+
+	it("routes to a page using a method POST", (done) => {
+		const req = {
+			method: "post",
+			protocol: "http",
+			secure: false,
+			hostname: "localhost",
+			url: "/postPage",
+		};
+		const expressRequest = new ExpressServerRequest(req);
+
+		requestContext.navigator.on('page', () => {
+			expect(expressRequest.getRouteName() === 'PostPage');
+			done();
+		});
+
+		requestContext.navigate(expressRequest, History.events.PAGELOAD);
+	});
+
+
+	it("routes to a page that can handle GET and POST using a method GET", (done) => {
+		const req = {
+			method: "get",
+			protocol: "http",
+			secure: false,
+			hostname: "localhost",
+			url: "/getAndPostPage",
+		};
+		const expressRequest = new ExpressServerRequest(req);
+
+		requestContext.navigator.on('page', () => {
+			expect(expressRequest.getRouteName() === 'GetAndPostPage');
+			done();
+		});
+
+		requestContext.navigate(expressRequest, History.events.PAGELOAD);
+	});
+
+	it("routes to a page that can handle GET and POST using a method POST", (done) => {
+		const req = {
+			method: "post",
+			protocol: "http",
+			secure: false,
+			hostname: "localhost",
+			url: "/getAndPostPage",
+		};
+		const expressRequest = new ExpressServerRequest(req);
+
+		requestContext.navigator.on('page', () => {
+			expect(expressRequest.getRouteName() === 'GetAndPostPage');
+			done();
+		});
+
+		requestContext.navigate(expressRequest, History.events.PAGELOAD);
+	});
+});
```

Comment:
Can you add some failure cases here too?  E.g. doesn't respond to a POST request when configured with PUT only &c.